languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml

```diff
@@ -34883,8 +34882,124 @@ USA
            </rule>
+           <rule>
+                <antipattern>
+                    <token skip="-1" chunk="E-NP-plural" />
+                    <token>this</token>
+                </antipattern>
+                <antipattern>
+                    <token postag="IN|RP|TO|V.*" postag_regexp="yes" />
+                    <token>this</token>
+                </antipattern>
+                <antipattern>
+                    <token>and</token>
+                    <token>this</token>
+                </antipattern>
+                <pattern>
+                    <token>this</token>
+                    <token min='0' postag_regexp='yes' postag='NN(:U|P)?'>
+                        <exception regexp='yes'>can|deal|one|google|excel|then|if|his|species|week|maybe</exception>
+                        <exception postag="W.*" postag_regexp="yes" />
+                        <exception regexp="yes">two|three|four|five|six|seven|eight|nine|ten|eleven|twelve</exception>
+                    </token>
+                    <token>have</token>
+                </pattern>
+                <message>The verb '\3' is plural.</message>
+                <suggestion>\1 \2 has</suggestion>
+                <suggestion>these <match no="2" postag="NNS" /> \3</suggestion>
+                <suggestion>\1 \2 had</suggestion>
+                <example correction="This girl has|These girls have|This girl had"><marker>This girl have</marker> beautiful eyes.</example>
+                <example>Southwest Prices in the San Juan Basin this summer have been left behind by rising California demand.</example>
+                <example>This is not functionality that we have built in or at this point have plans of adding.</example>
+                <example>The demons who started this have no idea about the kind of people they have taken on.</example>
+                <example>Fred and this customer have had a falling out.</example>
+            </rule>
+            <rule>
+                <antipattern>
+                    <token chunk_re="B-SBAR|B-PP" />
+                    <token chunk_re="B-NP.*" />
+                </antipattern>
+                <antipattern>
+                    <token chunk="E-NP-plural" />
+                    <token>that</token>
+                </antipattern>
+                <antipattern>
+                    <token postag="IN|RP|TO" postag_regexp="yes" />
+                    <token>that</token>
+                </antipattern>
+                <antipattern>
+                    <token>those</token>
+                    <token>of</token>
+                    <token postag="PRP" />
+                    <token>that</token>
+                </antipattern>
+                <antipattern>
+                    <token regexp="yes">and|see</token>
+                    <token>that</token>
+                </antipattern>
+                <pattern>
+                    <token>that</token>
+                    <token postag_regexp='yes' postag='NN|NN:U'>
```

Comment:
why not also "NN:UN"? (both uncountable and countable)

Reply:
I originally excluded it as it triggered an FP in rule [3], and I reused the code for [4]. However upon further inspection, the FP can be handled with a refined antipattern, so I’ve updated [3], [4], and [5] to include NN:UN. Also, I excluded NNP from [3] to be consistent with the other rules; including it brings up the issue surrounding company names (AKA collective nouns).